src/getters/getConfigFromChain.ts

```diff
@@ -0,0 +1,32 @@
+import { Contract } from 'ethers'
+import { StreamrConfig, streamrConfigABI } from '@streamr/network-contracts'
+import { getSigner } from '~/shared/stores/wallet'
+import { BNish } from '~/utils/bn'
+import { errorToast } from '~/utils/toast'
+import { defaultChainConfig } from './getChainConfig'
+
+/**
+ * When we will be needing more of the fields, simply expand the return type and add them
+ */
+export const getConfigFromChain = async (): Promise<{
+    maxPenaltyPeriodSeconds: BNish
+    minimumStakeWei: BNish
+} | null> => {
+    const signer = await getSigner()
+    const config = new Contract(
+        defaultChainConfig.contracts.StreamrConfig,
+        streamrConfigABI,
+        signer,
+    ) as StreamrConfig
+    try {
+        return {
+            maxPenaltyPeriodSeconds: await config.maxPenaltyPeriodSeconds(),
+            minimumStakeWei: await config.minimumStakeWei(),
+        }
+    } catch (e) {
+        errorToast({
+            title: 'Could not load data from the chain, please try again later.',
+        })
```

Comment:
Not sure if I like this toast. Later we throw if `getConfigFromChain` gives a `null`. If we decide to toast about it there then we have 2 toasts, one of which is uncontrollable from the outside.

Maybe drop the `try…catch`?

Reply:
Ok I'll try to improve it somehow. To be honest I'm kinda lacking some well-defined error-handling policy that we could be always referring to. We could speak about it on Monday

Reply:
What I did is I moved error handling "above" meaning that it's being handled wherever we are using this getter